community/proc/src/main/java/org/neo4j/proc/ProcedureSignature.java

```diff
@@ -126,6 +127,29 @@ public AnyType neo4jType()
         {
             return type;
         }
+
+        @Override
+        public String toString()
+        {
+            return String.format("%s :: %s", name, type);
+        }
+
+        @Override
+        public boolean equals( Object o )
+        {
+            if ( this == o ) { return true; }
+            if ( o == null || getClass() != o.getClass() ) { return false; }
+            FieldSignature that = (FieldSignature) o;
+            return name.equals( that.name ) && type.equals( that.type );
+        }
+
+        @Override
+        public int hashCode()
+        {
+            int result = name.hashCode();
+            result = 31 * result + type.hashCode();
```

Comment:
Missing test case here. Changing `31` to any other number does not break any tests. Changing the function to `return name.hashCode()` also does not break any tests.

Why `31` and a sum? Would be good with a comment to describe it in addition to the test cases.